cli/runtime/config/config_factory_test.go

```diff
@@ -0,0 +1,1245 @@
+}
+
+func TestGetClientConfigWithLockAndMigratedToNewConfig(t *testing.T) {
+	// Setup config test data
+	cfg, cfgNextGen, _, _ := setupCfgAndCfgNextGenData()
+	_, cleanUp := setupTestConfig(t, &CfgTestData{cfg: cfg, cfgNextGen: cfgNextGen, cfgMetadata: setupConfigMetadataWithMigrateToNewConfig()})
+
+	defer func() {
+		cleanUp()
+	}()
+
+	//Actions
+	node, err := getClientConfigNode()
+
+	// Assertions
+	assert.NotNil(t, node)
+	assert.NoError(t, err)
+
+	expectedCtx := &configapi.Context{
+		Name: "test-mc",
+		Type: configapi.CtxTypeK8s,
+		ClusterOpts: &configapi.ClusterServer{
+			Endpoint:            "test-endpoint",
+			Path:                "test-path",
+			Context:             "test-context",
+			IsManagementCluster: true,
+		},
+		DiscoverySources: []configapi.PluginDiscovery{
+			{
+				GCP: &configapi.GCPDiscovery{
+					Name:         "test",
+					Bucket:       "test-bucket",
+					ManifestPath: "test-manifest-path",
+				},
+			},
+			{
+				GCP: &configapi.GCPDiscovery{
+					Name:         "test-two",
+					Bucket:       "test-bucket",
+					ManifestPath: "test-manifest-path",
+				},
+			},
+		},
+	}
+
+	// Migrated To new config hence servers are not in the config-ng.yaml yet.
+	ctx, err := getContext(node, "test-mc")
+	assert.NoError(t, err)
+	assert.Equal(t, expectedCtx, ctx)
+
+	_, err = getServer(node, "test-mc")
+	assert.Equal(t, "could not find server \"test-mc\"", err.Error())
+}
+
+func TestGetClientConfigWithoutLockAndMigratedToNewConfig(t *testing.T) {
+	// Setup config test data
+	cfg, cfgNextGen, _, _ := setupCfgAndCfgNextGenData()
+	_, cleanUp := setupTestConfig(t, &CfgTestData{cfg: cfg, cfgNextGen: cfgNextGen, cfgMetadata: setupConfigMetadataWithMigrateToNewConfig()})
+
+	defer func() {
+		cleanUp()
+	}()
+
+	//Actions
+	AcquireTanzuConfigNextGenLock()
+	node, err := getClientConfigNodeNoLock()
+	ReleaseTanzuConfigNextGenLock()
+
+	// Assertions
+	assert.NotNil(t, node)
+	assert.NoError(t, err)
+
+	expectedCtx := &configapi.Context{
+		Name: "test-mc",
+		Type: configapi.CtxTypeK8s,
+		ClusterOpts: &configapi.ClusterServer{
+			Endpoint:            "test-endpoint",
+			Path:                "test-path",
+			Context:             "test-context",
+			IsManagementCluster: true,
+		},
+		DiscoverySources: []configapi.PluginDiscovery{
+			{
+				GCP: &configapi.GCPDiscovery{
+					Name:         "test",
+					Bucket:       "test-bucket",
+					ManifestPath: "test-manifest-path",
+				},
+			},
+			{
+				GCP: &configapi.GCPDiscovery{
+					Name:         "test-two",
+					Bucket:       "test-bucket",
+					ManifestPath: "test-manifest-path",
+				},
+			},
+		},
+	}
+
+	// Migrated To new config hence servers are not in the config-ng.yaml yet.
+	ctx, err := getContext(node, "test-mc")
+	assert.NoError(t, err)
+	assert.Equal(t, expectedCtx, ctx)
+
+	_, err = getServer(node, "test-mc")
+	assert.Equal(t, "could not find server \"test-mc\"", err.Error())
+}
+
+func TestPersistConfig(t *testing.T) {
+	// Setup config test data
```

Comment:
should you need to rebase, one more suggestion:

As implemented the persistConfig functionality will actually preserve any nodes in the main config.yaml that is not on the hardcode list. It would be good to have a test communicate that behavior.

perhaps something like:
```
--- a/cli/runtime/config/config_factory_test.go
+++ b/cli/runtime/config/config_factory_test.go
@@ -25,6 +25,7 @@ clientOptions:
         local:
           name: default-local
           path: standalone
+      # extra comment my local admin source
       - local:
           name: admin-local
           path: admin
@@ -99,6 +100,9 @@ contexts:
         contextType: tmc
 currentContext:
   k8s: test-mc
+# extraneous fiels/comments are preserved on file update
+extrafield:
+    extrasubfield: 1
 `
        expectedCfg := `apiVersion: config.tanzu.vmware.com/v1alpha1
 clientOptions:
@@ -110,6 +114,7 @@ clientOptions:
               local:
                 name: default-local
                 path: standalone
+            # extra comment my local admin source
             - local:
                 name: admin-local
                 path: admin
@@ -184,6 +189,9 @@ contexts:
           contextType: tmc
 currentContext:
     k8s: test-mc
+# extraneous fiels/comments are preserved on file update
+extrafield:
+    extrasubfield: 1
 `
```

Reply:
addressing  this in follow up PR